apps/expo/src/app/frlist/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { View, Text, TouchableOpacity, FlatList, ActivityIndicator, StyleSheet } from 'react-native';
+
+import { Route, router } from 'expo-router';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+
+import { useGlobalContext } from '~/context/global-context';
+import { api } from '~/utils/trpc/api';
+
+import SearchBar from '~/components/ui/search-bar/SearchBar'
+
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: '#1E1E1E',
+  },
+  friendItem: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    paddingHorizontal: 16,
+    paddingVertical: 12,
+  },
+  iconContainer: {
+    marginRight: 12,
+  },
+  username: {
+    fontSize: 16,
+    color: '#fff',
+  },
+  actionButton: {
+    paddingHorizontal: 8,
+    color: '#fff',
+  },
+});
+
+interface Friend {
+  id: number;
+  username: string;
+}
+
+const handleNavigateToProfile = (userId: number) => {
+  router.push(`/user/${userId}` as Route<string>);
+};
+
+const handleNavigateToMessages = async (friend: Friend) => {
+  const { userData } = useGlobalContext();
+  const mutation = api.friend.getChatIdByFriendId.useMutation();
+
+  // Use mutateAsync to perform the mutation and wait for the result
+  const chatId = await mutation.mutateAsync({
+    userId: userData?.id ?? 0,
+    friendId: friend.id,
+  });
+
+  if (chatId) {
+    router.push(`/messages/${chatId}` as Route<string>);
+  } else {
+    // Handle any error that may occur during chat creation
+    console.error('Failed to create a new chat');
+  }
+};
+
+const FriendsListScreen = () => {
+  const [friends, setFriends] = useState<Friend[]>([]);
+  const [filteredFriends, setFilteredFriends] = useState<Friend[]>([]);
+  const { userData } = useGlobalContext();
+  console.log('userData:', userData);
+
+  const { data: friendsData, isLoading, error } = api.friend.getFriends.useQuery(
+    { id: userData?.id ?? 0 },
+    { enabled: !!userData?.id }
```

Comment:
Enabled should be a boolean value, so instead of coercing a null or int into a boolean, just do a null check:

```ts
{ enabled: userData !== null }
```

Reply:
re-wrote index.tsx from merge with branch 220